### PR TITLE
Fix promtail alerts for mimir

### PR DESCRIPTION
**Pull request by QuentinBisson**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
Towards: https://github.com/giantswarm/roadmap/issues/3157

This PR fixes promtail alerts and make them work in mimir

### Checklist

- [ ] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/promtail.rules.yml [hunk 1]
```diff
@@ -14,7 +13,7 @@ spec:
           annotations:
             description: '{{`Scraping of all promtail pods to check if one failed every 5 minutes.`}}'
             opsrecipe: promtail-is-not-running/
-          expr: count(up{app="promtail"} == 0) > 0
+          expr: count(up{container="promtail"} == 0) > 0
```
Comment (marieroque) [inline helm/prometheus-rules/templates/alerting-rules/promtail.rules.yml:16]: Is `cluster_type="management_cluster"` missing ?

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/promtail.rules.yml:16]: No promtail should alert on all clusters

Comment (marieroque) [reply helm/prometheus-rules/templates/alerting-rules/promtail.rules.yml:16]: Even for customer promtail if they have one ?

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/promtail.rules.yml:16]: We do manage them so yes

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/promtail.rules.yml:16]: But it's mostly about those in kube-system

**Review by marieroque — APPROVED** [review]

**Comment by QuentinBisson** [comment]
If that pages too much then we can change that behavior @marieroque maybe splitting the promtail in kube-system and others, let;s see :)

**Pull request merged**